packages/components/src/DataList/components/DataListHeader/DataListHeader.test.tsx

```diff
@@ -0,0 +1,207 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import { DataListHeader } from "./DataListHeader";
+import { useDataListContext } from "../../context/DataListContext";
+import { useBatchSelect } from "../../hooks/useBatchSelect";
+import { useActiveLayout } from "../../hooks/useActiveLayout";
+import { useResponsiveSizing } from "../../hooks/useResponsiveSizing";
+
+// Mock the hooks
+jest.mock("../../context/DataListContext", () => ({
+  useDataListContext: jest.fn(),
+}));
+
+jest.mock("../../hooks/useBatchSelect", () => ({
+  useBatchSelect: jest.fn(),
+}));
+
+jest.mock("../../hooks/useActiveLayout", () => ({
+  useActiveLayout: jest.fn(),
+}));
+
+jest.mock("../../hooks/useResponsiveSizing", () => ({
+  useResponsiveSizing: jest.fn(),
+}));
+
+describe("DataListHeader", () => {
+  const mockUseDataListContext = useDataListContext as jest.Mock;
+  const mockUseBatchSelect = useBatchSelect as jest.Mock;
+  const mockUseActiveLayout = useActiveLayout as jest.Mock;
+  const mockUseResponsiveSizing = useResponsiveSizing as jest.Mock;
+
+  const mockLayout = jest.fn(data => (
+    <div data-testid="mock-header">{data.label}</div>
+  ));
+  const mockHeaders = { label: "Test Header" };
+
+  beforeEach(() => {
+    // Default mock settings
+    mockUseResponsiveSizing.mockReturnValue({ xs: true });
+
+    mockUseDataListContext.mockReturnValue({
+      headerVisibility: { xs: true, sm: true, md: true, lg: true, xl: true },
+      headers: mockHeaders,
+      layoutBreakpoints: ["xs"],
+      data: [{ id: 1 }, { id: 2 }],
+      selected: [],
+    });
+
+    mockUseBatchSelect.mockReturnValue({
+      hasAtLeastOneSelected: false,
+      canSelect: false,
+      canSelectAll: false,
+    });
+
+    mockUseActiveLayout.mockReturnValue({
+      layout: mockLayout,
+    });
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe("when headerVisibility is true", () => {
+    it("should render the header when layout is defined", () => {
+      render(<DataListHeader />);
+      expect(screen.getByTestId("mock-header")).toBeInTheDocument();
```

Comment:
I think using `toBeVisible` assertion might be a bit more useful when testing elements presence in the UI (so here, and in tests below):
```suggestion
      expect(screen.getByTestId("mock-header")).toBeVisible();
```